### PR TITLE
First set of RTL audit fixes

### PR DESCRIPTION
**Pull request by montehurd**
https://phabricator.wikimedia.org/T199569

<table>
  <tr>
    <td align=center>Before</td>
    <td align=center>After</td>
  </tr>
  <tr>
    <td colspan="2" align=center>Incorrect RTL padding</td>
  </tr>
  <tr>
    <td colspan="2">Save for later button</td>
  </tr>
  <tr>
    <td>
    screen shot 2018-07-25 at 1 34 27 pm
    </td>
    <td>
     screen shot 2018-07-25 at 1 33 31 pm
    </td>
  </tr>
  <tr>
    <td colspan="2">Announcement label</td>
  </tr>
  <tr>
    <td>
screen shot 2018-07-25 at 2 09 58 pm
    </td>
    <td>
screen shot 2018-07-25 at 2 08 18 pm
    </td>
  </tr>
  <tr>
    <td colspan="2">Picture label<br>3ba9538 and 05c953c made the label consistently use the full available width<br>f5b5117 made the text use correct alignment (for some reason `.natural` didn't seem to work...)</td>
  </tr>
  <tr>
    <td>
screen shot 2018-07-26 at 12 18 37 pm
    </td>
    <td>
screen shot 2018-07-26 at 12 10 13 pm
    </td>
  </tr>
</table>

### CODE REVIEW

File: Wikipedia/Code/ArticleFullWidthImageCollectionViewCell.swift [hunk 1]
```diff
@@ -92,7 +92,7 @@ open class ArticleFullWidthImageCollectionViewCell: ArticleCollectionViewCell {
 
         if !isSaveButtonHidden {
             origin.y += spacing - 1
-            let saveButtonFrame = saveButton.wmf_preferredFrame(at: origin, maximumWidth: widthMinusMargins, horizontalAlignment: isDeviceRTL ? .right : .left, apply: apply)
+            let saveButtonFrame = saveButton.wmf_preferredFrame(at: origin, maximumWidth: size.width, horizontalAlignment: isDeviceRTL ? .right : .left, apply: apply)
```
Comment (joewalsh) [inline Wikipedia/Code/ArticleFullWidthImageCollectionViewCell.swift]: This should stay `widthMinusMargins`, the bug is in `UIView+SemanticContent.swift`:

```
extension AlignedImageButton {
    override public func wmf_preferredFrame(at point: CGPoint, maximumSize: CGSize = NoIntrinsicSize, minimumSize: CGSize = NoIntrinsicSize, horizontalAlignment: HorizontalAlignment = .center, verticalAlignment: VerticalAlignment = .center, apply: Bool = false) -> CGRect  {
        let adjustedPoint = CGPoint(x: point.x - leftPadding, y: point.y - verticalPadding)
        var adjustedSize = maximumSize
        if adjustedSize.width != UIViewNoIntrinsicMetric {
            adjustedSize.width = adjustedSize.width + leftPadding + rightPadding
        }
        return super.wmf_preferredFrame(at: adjustedPoint, maximumSize: maximumSize, minimumSize: minimumSize, horizontalAlignment: horizontalAlignment, verticalAlignment: verticalAlignment, apply: apply)
    }
}
```

Comment (montehurd) [reply Wikipedia/Code/ArticleFullWidthImageCollectionViewCell.swift]: oh cool thx!

Comment (joewalsh) [reply Wikipedia/Code/ArticleFullWidthImageCollectionViewCell.swift]: It pushes the frame to account for the extra button space, it's hardcoded for .left and .top alignment, it needs to be updated to respect the other alignments

Comment (montehurd) [reply Wikipedia/Code/ArticleFullWidthImageCollectionViewCell.swift]: turns out we weren't using the `adjustedSize` in that override

Comment (montehurd) [reply Wikipedia/Code/ArticleFullWidthImageCollectionViewCell.swift]: Looks better now: 

| Before  | After |
| ------------- | ------------- |
| screen shot 2018-07-25 at 1 34 27 pm | screen shot 2018-07-25 at 1 33 31 pm |

**Comment by montehurd** [comment]
@joewalsh @nambatee 
the comments i added to the screenshot table may be useful context for the last few commits i just pushed... i may have missed easier ways to fix these... lemme know!

**Comment by montehurd** [comment]
oops the last 4 before/after images were reversed - just fixed...

File: Wikipedia/Code/WMFFeedImage.m [hunk 2]
```diff
@@ -10,6 +11,7 @@ @implementation WMFFeedImage
 + (NSDictionary *)JSONKeyPathsByPropertyKey {
     return @{WMF_SAFE_KEYPATH(WMFFeedImage.new, canonicalPageTitle): @"title",
              WMF_SAFE_KEYPATH(WMFFeedImage.new, imageDescription): @"description.text",
+             WMF_SAFE_KEYPATH(WMFFeedImage.new, imageDescriptionIsRTL): @"description.lang",
              WMF_SAFE_KEYPATH(WMFFeedImage.new, imageURL): @"image.source",
```
Comment (joewalsh) [inline Wikipedia/Code/WMFFeedImage.m]: I think you have to bump the model version by adding
```+ (NSUInteger)modelVersion {
    return 2;
}``` 
when you update the model. We should also verify that this continues to work with old cached data after this change.

**Pull request merged**